### DISCUSSION TITLE
[Protokoll] 16.04.2024

### DISCUSSION THREAD

**Discussion by 7ben18** [AdversarialAttacks/.github#23] [Protokolle]
### Wer hat an dieser Sitzung teilgenommen?

- [X] Gabriel Torres Gamez
- [X] Si Ben Tran
- [ ] Daniel Perruchoud
- [X] Stephan Heule

### Welche sind die wichtigsten Themen, die an dieser Sitzung besprochen wurden?

Wir haben hauptsächlich das Training der Modelle und die Ergebnisse besprochen. Wir sind kurz noch auf das weitere Vorgehen mit Perturbationen eingegangen und Stephan über den aktuellen Status des Zugriffs auf Slurm oder Alternativen eingegangen.

### Wo haben wir Hindernisse? Was blockiert den Fortschritt unser Projekts?

Trotz Gradient Clipping, Logits Clipping und Gradual Unfreezing konnten wir das Alexnet und das VGG nicht zum lernen bringen. Jedoch sollen wir jetzt nicht mehr Zeit in das Training dieser Modelle stecken, da wir genügend Modelle haben, welche funktionieren und uns eher auf die Perturbationen fokussieren müssen. Es ist komisch, dass das Training bei den anderen Modellen ausser diesen beiden funktioniert. 

### Was wurde an der Sitzung beschlossen?

- Wir haben genug Modelle, an welchen wir die Perturbationen testen können.
- Wir sollten die Ergebnisse unsere Modelle testen, dokumentieren und visualisieren, damit wir sie nächste Woche mit Dani besprechen können. Zusatz: Wir sollten noch testen, wie sich das Modell verhaltet, wenn wir den Threshold verändern.
- Wir fahren jetzt auch mit dem neuen Meilenstein (Erstellung von Petrurbationen) fort. Dafür sollten wir zuerst auf ein Modell fokusieren und testen, dann auf die anderen erweitern.

**Comment by gabrieltorresgamez** [comment]
#### Modell Training
Validierungs- AUC, Recall, Specificity von Modelltraining dem Stephan gezeigt. <br>Siehe Notebook: [04-evaluation.ipynb](https://github.com/AdversarialAttacks/main/blob/main/notebooks/04-evaluation.ipynb)

Wichtige Bemerkung: In Weights und Bias loggen wir den letzten und besten Runs aufgrund des Validierungsloss. 

Die Plots im Notebook zeigen die besten Runs pro Modell pro Datensatz. 
Im mri_data sind weniger Modelle vertreten, da wir mittels Timeboxing das Zeitkontingente für gewisse Modellfamilien erreicht haben. 

Modelle die funktioniert haben und erfolgreich trainiert wurden können wir zu unserem Pool von Modellen hinzufügen die wir für die Perturbationen verwenden können. 

#### Dokumentation von Modelltraining
Die Dokumentation von AlexNet und VGG  (Nicht erfolgreiches Training) kann als Anhang, Bericht, oder komplett weg gelassen werden. Details können nächste Woche im Plenum besprochen werden, da diese noch nicht relevant sind. 

In der nächsten Weekly Sitzung Visualisierungen mit **Test** Performance Plots (Konfusionsmatrix, oder Barplots) zeigen. 
Aus dem ersten Meilenstein die Key-Points definieren, welche in den Bericht einfliessen. 

Nachtrag: 
![image](https://github.com/AdversarialAttacks/.github/assets/66916399/e699da38-10dc-409e-871b-5123d4c36556)


#### Perturbation
Für den nächsten Meilenstein (Start neuer Meilenstein am 17.04.2024) legen wir den Fokus auf das erstellen von Universelle Perturbationen. 

Die Perturbationen sollen wir zuerst mit einem "mittelschweres" Modell aus unseren Modellsack wählen und ein Proof of Konzept durchführen und an einem Beispiel durch exerzieren. Diese können wir dann jeweils im Plenum besprechen und diskutieren in den jeweiligen Weeklys. 

#### Rechenkosten
Rechenkosten werden aktuell von Studierenden getragen. Aktuell nutzen wir Lamda Cloud und Gabriel und Bens Grafikkarten. 
Informationen folgen noch.
